### PR TITLE
Change attributes field to use JSONField

### LINKED ISSUE #7: `ProjectDataset.data` should be a `JSONField`
Pickle is Python-specific and not queryable/indexable.

### PR DESCRIPTION
**Pull request by baali**
closes #7

### CODE REVIEW

File: bw_projects/filesystem.py [hunk 1]
```diff
@@ -1,30 +1,48 @@
...
 
 
-def create_dir(dirpath):
+re_slugify = re.compile(r"[^\w\s-]", re.UNICODE)
+SUBSTITUTION_RE = re.compile(r"[^\w\-\.]")
+MULTI_RE = re.compile(r"_{2,}")
+
+
+def safe_filename(
+    string: Union[str, bytes], add_hash: bool = True, full: bool = False
```
Comment (cmutel) [inline bw_projects/filesystem.py:16]: Maybe this is covered elsewhere, but I would drop`full` and just make it always effectively `False`.

File: bw_projects/project.py [hunk 2]
```diff
@@ -7,17 +7,16 @@
...
-from .filesystem import create_dir, maybe_path
-from .sqlite import PickleField, SubstitutableDatabase
+from .filesystem import create_dir, maybe_path, safe_filename
+from .sqlite import JSONField, SubstitutableDatabase
 
 
 class ProjectDataset(Model):
-    data = PickleField()
     name = TextField(index=True, unique=True)
+    attributes = JSONField()
```
Comment (cmutel) [inline bw_projects/project.py:19]: I like this idea, but we need to make sure that this is clear in the documentation, and is marked as a breaking change for existing Brightway users. I don't think it should effect many users, but better safe than sorry.

File: bw_projects/filesystem.py (hunk 1 above)
Comment (baali) [reply bw_projects/filesystem.py:16]: Thank you for the suggestion, I will include this suggestion in [other PR](https://github.com/baali/bw_projects/pull/13) where I have introduced this function.

I rebased this PR on another open one and that's why the confusion.

File: bw_projects/sqlite.py [hunk 3]
```diff
@@ -1,14 +1,23 @@
-import pickle
+import json
 
-from peewee import BlobField, SqliteDatabase
+from peewee import TextField, SqliteDatabase
 
 
-class PickleField(BlobField):
+class JSONField(TextField):
```
Comment (cmutel) [inline bw_projects/sqlite.py:6]: Please consider using the [native JSON field](https://docs.peewee-orm.com/en/latest/peewee/sqlite_ext.html#sqlite-json1); not sure if it makes more sense, but we should add a line about why we are rejecting it (if we are).

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,2 @@
 appdirs
-bw_processing
```
Comment (cmutel) [inline requirements.txt:2]: Nice!

File: bw_projects/sqlite.py (hunk 3 above)
Comment (baali) [reply bw_projects/sqlite.py]: Thank you @cmutel for this pointer, yes using the native field is better. Right now, I don't think we have specific requirement that is not covered by the native field.

I have now shifted the code to use this field.

**Pull request merged**